Firebase/Core/FIRApp.m

```diff
@@ -513,6 +496,21 @@ + (void)registerLibrary:(nullable id<FIRLibrary>)library
   }
 }
 
++ (void)registerInternalLibrary:(nonnull Class<FIRLibrary>)library
+                       withName:(nonnull NSString *)name
+                    withVersion:(nonnull NSString *)version {
+  // This is called at +load time, keep the work to a minimum.
```

Comment:
I notice that we check if `library` conforms to `FIRLibrary` down below in the `registerAsComponentRegistrant` method. It should probably move here in case we add another required method and make assumptions.

Reply:
Done. It also solves the need to come back later and fix a TODO. Thanks!

I also moved test only code from FirebaseCore to the tests.